src/Microsoft.AspNet.Mvc.Core/UrlActionContext.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNet.Mvc
+{
+    /// <summary>
+    /// Context object to be used for the URLs that <see cref="Mvc.IUrlHelper.Action(UrlActionContext)"/> generates.
+    /// </summary>
+    public class UrlActionContext
+    {
+        /// <summary>
+        /// The name of the action method that <see cref="Mvc.UrlHelper.Action(UrlActionContext)" uses to generate URLs.
+        /// </summary>
+        public string Action
+        {
+            get;
+            set;
+        }
+
+        /// <summary>
+        /// The name of the controller that <see cref="Mvc.UrlHelper.Action(UrlActionContext)" uses to generate URLs.
+        /// </summary>
+        public string Controller
```

Comment:
How about ActonName, ControllerName, or RouteValues instead of Action, Controller, Values??